lib/index.js

```diff
@@ -82,7 +82,10 @@ function S3(uri, callback) {
     }
 
     var source = this;
-    if (uri.data) { return setup(null, uri.data); }
+    if (uri.data) {
+      if (uri.strict && uri.strict === true) uri.data.strict = true;
```

Comment:
Rather than attaching this to `uri.data` could it just be assigned to `source.strict` in the `setup()` function?